src/bisection.cu

```diff
@@ -0,0 +1,121 @@
+/*************************************************************************
+ * Copyright (c) 2016-2022, NVIDIA CORPORATION. All rights reserved.
+ *
+ * See LICENSE.txt for license information
+ ************************************************************************/
+
+#include "cuda_runtime.h"
+#include "common.h"
+
+void BisectionGetCollByteCount(size_t *sendcount, size_t *recvcount, size_t *paramcount, size_t *sendInplaceOffset, size_t *recvInplaceOffset, size_t count, int nranks) {
+  *sendcount = count;
+  *recvcount = count;
+  *sendInplaceOffset = 0;
+  *recvInplaceOffset = 0;
+  *paramcount = count;
+}
+
+int getPeer(int rank, int n_ranks){
+    if (n_ranks % 4 == 0)
```

Comment:
Why is n_ranks % 4 == 0 a special case?

Reply:
This allow cross rail bisection for some clusters with rail-optimized topology, for example IL1